feat: get ready for OSS, updating licenses and CI

ayuskauskas opened the pull request.

ayuskauskas committed:
feat: print configuration at the start of agent container

ayuskauskas committed:
docs(agent): add notice file and remove extra license file

ayuskauskas committed:
feat: change to common license formatter and update all code with that format

fix:(operator): Change operator-container to only go if ci works
feat(operator): link top level license here so license-check doesn't fail

ayuskauskas committed:
feat: consolidate agent and operator CI to better control dependencies

Fix license formatter to recognize when license hasn't changed and not introduce extra lines

ayuskauskas committed:
feat(operator): rename the ci workflow to match agent workflow

lockwobr reviewed (changes requested):
The help chart license maybe should also be a symlink?

ayuskauskas committed:
fix: miscellaneous fixes to project structure

Make chart license file a symlink of top level one
Add a top level .gitignore
Remove boilerplate references from operator mockery file
Remove .vscode directory
